### PR TITLE
Add release info to the metadatafile for AQA-Cert

### PR DESCRIPTION
**Pull request by JeromeJu**
Add release info for AQA-Cert

Fixes: https://github.com/adoptium/TKG/issues/260

Signed-off-by: Jerome Ju [jeromeqju@gmail.com](mailto:jeromeqju@gmail.com)

### CODE REVIEW

File: src/org/openj9/envInfo/EnvDetector.java [hunk 1]
```diff
@@ -72,6 +74,7 @@ private static void getJavaInfo() {
 			output.write(JDKVERSIONvalue);
 			output.write(JDKIMPLvalue);
 			output.write(JDKVENDORvalue);
+			output.write(ReleaseInfo); // TOOD: Add to "AQACert.log" file when rebase
```
Comment (llxia) [inline src/org/openj9/envInfo/EnvDetector.java:77]: This cannot be output into autoGenEnv.mk. This info should be AQACert.log.
We should wait and get PR https://github.com/adoptium/TKG/pull/266/ completed first.

Comment (JeromeJu) [reply src/org/openj9/envInfo/EnvDetector.java:77]: Would like to check the workflow here as it's my first time working in a forked repo.

My thoughts are:
1. Cherrypick the merged `260_release change` that's merged later yesterday
2. Rease on the `260_release change`

Please let me know if I'm mistaken, thanks!

Comment (JeromeJu) [reply src/org/openj9/envInfo/EnvDetector.java]: Used github client to resolve

**Comment by JeromeJu** [comment]
Latest Grinder link: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3654/

Also would like to ask on which platform may contains the release info file path? Thanks

**Comment by smlambert** [comment]
re: https://github.com/adoptium/TKG/pull/272#issuecomment-1045529211 Every hotspot build would have a release file (for every platform).

**Comment by JeromeJu** [comment]
Thanks @smlambert!
As discussed and pointed during the meet, I've been trying to locate the release info under {JAVA_HOME}/build/release:
https://github.com/JeromeJu/TKG/blob/release_test_1/src/org/openj9/envInfo/JavaInfo.java#L167-L176 should be correct referring to https://github.com/adoptium/aqa-tests/blob/master/get.sh#L586-L601.

Following up on that, I've been trying to look into the directory under {JAVA_HOME} for hotspot,  and it didn't seem to have the /build/release path according to [Grinder 3667](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3667/)
https://github.com/JeromeJu/TKG/blob/release_test_1/src/org/openj9/envInfo/JavaInfo.java#L178-L184

**Comment by smlambert** [comment]
re: https://github.com/adoptium/TKG/pull/272#issuecomment-1046258663 - as part of your investigation, you could download a build from https://adoptium.net/ , unpack it locally to find the release file (to be familiar with the directory structure).

When I do so for mac when I unpack the build, the release file is not in a /build directory, but at the top-level of JAVA_HOME.

**Comment by llxia** [comment]
From a quick look at your Grinder, 
- I see the release info got printed out in shell script in [Grinder 3667](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3667/). It means the file is there. Please check your java code.

image

- your branch is out of date. I still see the old error in `java -version`:

image

**Comment by JeromeJu** [comment]
Thanks for the guidance!

1. I've updated the current PR
2. As pointed for the build, I've also followed the `JAVA_HOME/release`, however, I wasn't able to access 'JAVA_HOME/release'
3. Therefore, I 'ls' through 'JAVA_HOME' to investigate, and got the following results:  [#3713](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3713/artifact/aqa-tests/TKG/AQACert.log)
```
DETECTED_JAVAHOME_FILES=ASSEMBLY_EXCEPTION
LICENSE
THIRD_PARTY_README
bin
lib
```
4. Also when I remove the check for the 'JAVA_HOME/release', it shows: [#3704](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3704/artifact/aqa-tests/TKG/AQACert.log)
```
DETECTED_RELASE_INFO=cat: /home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image/jre/release: No such file or directory
```

**Comment by JeromeJu** [comment]
The printed `JAVA_HOME` is `/home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image/jre`

**Comment by smlambert** [comment]
Right, so the release file you are looking for will be up one level from that location.  The pseudocode would be "If javaHome string endsWith(jre), then strip it off to get the path to the file".  It may not always be true that JAVA_HOME will point to jre dir (as in some versions of java, it may not be present).  You will have to run tests with a few different JDK_VERSION in Grinders, like JDK_VERSION=18.

**Comment by llxia** [comment]
I think we can avoid jre issue by using `TEST_JDK_HOME`. Could you try to get `TEST_JDK_HOME`?
```
String TEST_JDK_HOME = System.getenv("TEST_JDK_HOME");
```

**Comment by JeromeJu** [comment]
Thanks @llxia and @smlambert, according to [Grinder link](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3733/), TEST_JDK_HOME shall fix the path for `java.home`.

File: src/org/openj9/envInfo/EnvDetector.java [hunk 2]
```diff
@@ -61,6 +62,7 @@ private static void getJavaInfo() {
 		String JDKIMPLvalue = "DETECTED_JDK_IMPL=" + javaImplInfo + "\n";
 		String JDKVENDORvalue = "DETECTED_JDK_VENDOR=" + vendorInfo + "\n";
 		String JavaVersionValue = "DETECTED_JAVA_VERSION=" + javaVersion + "\n";
+		String ReleaseValue = "DETECTED_RELASE_INFO=" + releaseInfo + "\n";
```
Comment (smlambert) [inline src/org/openj9/envInfo/EnvDetector.java:65]: please fix the spelling of DETECTED_RELASE_INFO to DETECTED_RELEASE_INFO

File: src/org/openj9/envInfo/JavaInfo.java [hunk 3]
```diff
@@ -164,6 +164,18 @@ public String getJDKVendor() {
         }
     }
 
+    public String getReleaseInfo() {
+        String rt = "";
+        CmdExecutor ce = CmdExecutor.getInstance();
+        String TEST_JDK_HOME = System.getenv("TEST_JDK_HOME");
```
Comment (smlambert) [inline src/org/openj9/envInfo/JavaInfo.java:170]: Please change the variable name to use the typical standard for naming java variables, lowerCamelCase for these types of local variables.  All caps ([SCREAMING_SNAKE_CASE](https://en.wikipedia.org/wiki/Snake_case)) is reserved for variables denoted as static final keywords.

**Review by smlambert — COMMENTED** [review]
thanks @JeromeJu, getting there!  Please see my change requests for this PR inline

**Comment by JeromeJu** [comment]
Rerun [Grinder](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3734/) after changing the naming and the typo! Please let me know if it looks fine. Thank you for the pointers!

**Review by llxia — APPROVED** [review]
LGTM

**Review by smlambert — APPROVED** [review]
thanks @JeromeJu !

**Pull request merged**